lib/portlayer/network/context.go

```diff
@@ -858,6 +858,94 @@ func (c *Context) container(h *exec.Handle) (*Container, error) {
 	return nil, ResourceNotFoundError{error: fmt.Errorf("container %s not found", id.String())}
 }
 
+func (c *Context) populateAliases(con *Container, s *Scope, e *Endpoint) []string {
+	var aliases []string
+
+	// aliases to remove
+	// name for dns lookup
+	aliases = append(aliases, fmt.Sprintf("%s:%s", s.Name(), con.name))
+	aliases = append(aliases, fmt.Sprintf("%s:%s", s.Name(), con.id.Truncate()))
+	for _, as := range e.aliases {
+		for _, a := range as {
+			aliases = append(aliases, a.scopedName())
+		}
+	}
+
+	// aliases from other containers
+	for _, e := range s.Endpoints() {
+		if e.Container() == con {
+			continue
+		}
+
+		for _, a := range e.getAliases(con.name) {
+			aliases = append(aliases, a.scopedName())
+		}
+	}
+	return aliases
+}
+
+func (c *Context) removeAliases(aliases []string, con *Container) {
+	// remove aliases
+	for _, a := range aliases {
+		as := c.aliases[a]
+		for i := range as {
+			if as[i] == con {
+				as = append(as[:i], as[i+1:]...)
+				if len(as) == 0 {
+					delete(c.aliases, a)
+				} else {
+					c.aliases[a] = as
+				}
+
+				break
+			}
+		}
+	}
+	// long id
+	delete(c.containers, con.ID().String())
+	// short id
+	delete(c.containers, con.ID().Truncate().String())
+	// name
+	delete(c.containers, con.Name())
+}
+
+func (c *Context) UnbindContainerByID(id string) ([]*Endpoint, error) {
```

Comment:
Should we call this RemoveIDFromScopes or similar? This isn't really an Unbind as we're not updating the container, just the tracking data for it.
It's not really evident unless looking at the handleEvent code what the difference is and definitely not why.

Reply:
Done